accumulator_updater/programs/accumulator_updater/src/instructions/create_inputs.rs

```diff
@@ -0,0 +1,135 @@
+use {
+    crate::{
+        accumulator_acc_seeds,
+        accumulator_acc_seeds_with_bump,
+        state::*,
+        AccumulatorUpdaterError,
+    },
+    anchor_lang::{
+        prelude::*,
+        system_program::{
+            self,
+            CreateAccount,
+        },
+    },
+};
+
+
+pub fn create_inputs<'info>(
+    ctx: Context<'_, '_, '_, 'info, CreateInputs<'info>>,
+    base_account: Pubkey,
```

Comment:
Is this base account field necesary?

Reply:
Why not use the caller program as the base

Reply:
the caller program is included as part of the seeds for the PDAs that are created. We need the `base_account` to distinguish between the PDAs generated for the same account type/schema e.g.  `PriceFullMessage` for BTC/USD vs ETH/USD. This could be something other than the `base_account` pubkey but this seemed to be the most natural fit.

Reply:
ah got it